components/raftstore/src/store/fsm/peer.rs

```diff
@@ -774,103 +754,81 @@ where
         }
-
-        {
-            let mut meta = self.ctx.store_meta.lock().unwrap();
-            meta.set_region(
-                &self.ctx.coprocessor_host,
-                region.clone(),
-                &mut self.fsm.peer,
+	let to_be_removed: Vec<metapb::Peer> = self
+            .region()
+            .get_peers()
+            .iter()
+            .filter(|&peer| !new_peer_list.contains(&peer.get_id()))
+            .map(|peer| peer.clone())
+            .collect();
+        for peer in to_be_removed {
+            let mut req = new_admin_request(region.get_id(), self.fsm.peer.peer.clone());
+            let mut admin = AdminRequest::default();
+            admin.set_cmd_type(AdminCmdType::ChangePeer);
+            admin
+                .mut_change_peer()
+                .set_change_type(ConfChangeType::RemoveNode);
+            admin.mut_change_peer().set_peer(peer.clone());
+            req.set_admin_request(admin);
+            self.propose_raft_command_internal(
+                req,
+                Callback::None,
+                DiskFullOpt::AllowedOnAlmostFull,
             );
-            if meta
-                .region_ranges
-                .remove(&enc_end_key(original_region_state.get_region()))
-                .is_none()
-            {
-                panic!(
-                    "{} original region does not exist in store meta",
-                    self.fsm.peer.tag
-                );
-            }
-            for (_, id) in meta.region_ranges.range((
-                Excluded(keys::data_key(region.get_start_key())),
-                Unbounded::<Vec<u8>>,
-            )) {
-                let exist_region = &meta.regions[id];
-                if enc_start_key(exist_region) >= keys::data_end_key(region.get_end_key()) {
-                    break;
-                }
-                panic!(
-                    "{:?} is overlapped with an existing region {:?}",
-                    region, exist_region
-                );
-            }
-            if meta
-                .region_ranges
-                .insert(enc_end_key(&region), region.get_id())
-                .is_some()
-            {
-                panic!(
-                    "key conflicts while inserting region {:?} into store meta",
-                    region
-                );
-            }
         }
-        for peer_id in to_be_removed.clone() {
-            let mut cc = eraftpb::ConfChangeV2::default();
-            let mut ccs = eraftpb::ConfChangeSingle::default();
-            ccs.set_change_type(eraftpb::ConfChangeType::RemoveNode);
-            ccs.set_node_id(peer_id);
-            cc.set_transition(eraftpb::ConfChangeTransition::Auto);
-            cc.mut_changes().push(ccs);
-            if let Err(e) = self.fsm.peer.raft_group.apply_conf_change(&cc) {
-                panic!("fail to apply conf change for unsafe recover {:?}", e);
+
+        // forward commit index.
+        self.fsm.peer.raft_group.raft.raft_log.committed =
```

Comment:
no need to forward commit index, it should be performed in force leader state

Reply:
Forced leader only makes the logs to be able to replicated, but cannot help committing new entries due to lack of quorum, right?

Reply:
force leader state will help you to forward the commit index

Reply:
you can rebase this PR on the force leader PR, so you can run though the test without this